lib/ansible/modules/system/mount.py

```diff
@@ -521,14 +521,8 @@ def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
     """Gather mount information"""
 
     try:
-        f = open(mntinfo_file)
-    except IOError:
-        return
-
-    lines = map(str.strip, f.readlines())
-
-    try:
-        f.close()
+        with open(mntinfo_file) as f:
+            lines = map(str.strip, f.readlines())
     except IOError:
         module.fail_json(msg="Cannot close file %s" % mntinfo_file)
```

Comment:
Also, this error message is now wrong.

Reply:
Yes. I did not notice that :( As written above, converting it to a context manager may cause some slight semantic changes. Python 2 makes it cumbersome to catch the file not found error (it requires errno and I'm not sure if it's enough to check `exception.errno == errno.ENOENT`). I think that some of the explicit open error checking and error with closing the file should be reconsidered, and if they can be merged.

Reply:
How about splitting up the PR into multiple ones? At least one for the cases where semantics have to change (probably best one PR per module, because it will need discussion for every one of them, and having one module per PR increases the chance that some of them will be merged, even if some others are not or take forever), and one for the "obvious" changes.

Reply:
Yes. I think that'd be the obvious outcome of some of these discussions. Thanks for the suggestions :) Ansible seems to be somewhat conservative about changes, for good reason, considering that it's in use by quite many people, and in very diverse contexts.